webapp/channels/src/packages/mattermost-redux/src/reducers/entities/scheduled_posts.ts

```diff
@@ -65,6 +65,30 @@ function byTeamId(state: ScheduledPostsState['byTeamId'] = {}, action: AnyAction
 
         return modified ? newState : state;
     }
+    case ScheduledPostTypes.SCHEDULED_POST_DELETED: {
+        const scheduledPost = action.data.scheduledPost;
+
+        const newState = {...state};
+        let modified = false;
+
+        Object.keys(state).some((teamId: string) => {
+            const index = newState[teamId].findIndex((existingScheduledPost) => existingScheduledPost.id === scheduledPost.id);
+
+            if (index >= 0) {
+                newState[teamId] = [...newState[teamId]];
+                newState[teamId].splice(index, 1);
+                modified = true;
+
+                // return true makes some() not loop through remaining array
```

Comment:
I appreciate the comment explaining why this behaves this way, but I think this might be easier to write using something like `for (const teamId of Object.keys(state)) {...}` (or that but using `Object.entries`) so that we can just break out of the loop once the object is found. Thoughts?

Reply:
Done.